### PR TITLE
🏗♻️ Consolidate all `babel` configs into `babel.config.js` (with `--esm` fixes)

### PR DESCRIPTION
**Pull request by rsimha**
**Background:**
With #27576 and #27664, all the `babel` transform configs in `build.conf.js` and `babel.config.js` were consolidated into `build-system/babel-config/`. See https://github.com/ampproject/amphtml/pull/27576#issue-398876223 for a description of the refactor, and https://github.com/ampproject/amphtml/pull/27576#issuecomment-611384227 for a description of the tests that were run.

**Reason for revert:**
#27576 and #27664 had to be reverted in #27678 because they broke `gulp dist --esm`. @kristoferbaxter investigated the break via #27676 and discovered the following:

- Before #27576, not all files were being compiled in esm mode even when the `--esm` flag was used.
- With #27576, all files were being compiled in esm mode with `--esm` (as desired).
- However, this broke `amp-viewer-host`, which used `compilationLevel: 'WHITESPACE_ONLY'`.
- In addition, babel presets were being run during post-closure instead of pre-closure during esm builds.

**This PR does the following.**
- Reverts #27678 (and thereby reinstates #27576 and #27664)
- Removes the compilation level override for `amp-viewer-host` (it now uses closure's default)
- Moves presets for esm mode from post-closure to pre-closure

**With this PR:**
- Compilation output remains unchanged from for unminfied and regular minified output. See https://github.com/ampproject/amphtml/pull/27576#issuecomment-611384227.
- The full `gulp dist --esm` build has been fixed.

**Bundle sizes for the esm build:**
```
rsimha@rsimha-macbookpro2 (2020-04-09-BabelRefactorRedux):~/src/amphtml$ gulp bundle-size --on_local_build
[19:39:25] Using gulpfile ~/src/amphtml/gulpfile.js
[19:39:25] Starting 'bundle-size'...
[19:39:25] Computing bundle size for version 2004092326140 at commit 39061a2.
[19:39:25] brotli bundle sizes are:
[19:39:25]   dist/alp.mjs: 5.73KB
[19:39:25]   dist/amp-story-player-v0.mjs: 4.06KB
[19:39:25]   dist/amp4ads-host-v0.mjs: 6.81KB
[19:39:25]   dist/amp4ads-v0.mjs: 50.51KB
[19:39:25]   dist/examiner.mjs: 0.36KB
[19:39:26]   dist/shadow-v0.mjs: 61.96KB
[19:39:26]   dist/v0.mjs: 61.97KB
[19:39:26]   dist/video-iframe-integration-v0.mjs: 1.96KB
[19:39:26]   dist/ww.mjs: 13.41KB
[19:39:26]   dist/v0/amp-3d-gltf-0.1.mjs: 5.76KB
[19:39:26]   dist/v0/amp-3q-player-0.1.mjs: 8.04KB
[19:39:26]   dist/v0/amp-access-0.1.mjs: 15.06KB
[19:39:26]   dist/v0/amp-access-laterpay-0.1.mjs: 3.77KB
[19:39:26]   dist/v0/amp-access-laterpay-0.2.mjs: 3.88KB
[19:39:26]   dist/v0/amp-access-poool-0.1.mjs: 3.52KB
[19:39:26]   dist/v0/amp-access-scroll-0.1.mjs: 5.13KB
[19:39:26]   dist/v0/amp-accordion-0.1.mjs: 5.01KB
[19:39:26]   dist/v0/amp-action-macro-0.1.mjs: 1.75KB
[19:39:26]   dist/v0/amp-ad-0.1.mjs: 17.11KB
[19:39:27]   dist/v0/amp-ad-custom-0.1.mjs: 52.69KB
[19:39:27]   dist/v0/amp-ad-exit-0.1.mjs: 5.46KB
[19:39:27]   dist/v0/amp-ad-network-adsense-impl-0.1.mjs: 66.79KB
[19:39:27]   dist/v0/amp-ad-network-adzerk-impl-0.1.mjs: 61.28KB
[19:39:27]   dist/v0/amp-ad-network-doubleclick-impl-0.1.mjs: 75.41KB
[19:39:28]   dist/v0/amp-ad-network-fake-impl-0.1.mjs: 62.39KB
[19:39:28]   dist/v0/amp-addthis-0.1.mjs: 7.51KB
[19:39:28]   dist/v0/amp-analytics-0.1.mjs: 26.61KB
[19:39:28]   dist/v0/amp-anim-0.1.mjs: 1.2KB
[19:39:28]   dist/v0/amp-animation-0.1.mjs: 27.19KB
[19:39:28]   dist/v0/amp-apester-media-0.1.mjs: 9.26KB
[19:39:28]   dist/v0/amp-app-banner-0.1.mjs: 3.81KB
[19:39:28]   dist/v0/amp-audio-0.1.mjs: 3.65KB
[19:39:28]   dist/v0/amp-auto-ads-0.1.mjs: 4.65KB
[19:39:28]   dist/v0/amp-auto-lightbox-0.1.mjs: 2.42KB
[19:39:28]   dist/v0/amp-autocomplete-0.1.mjs: 8.61KB
[19:39:28]   dist/v0/amp-base-carousel-0.1.mjs: 8.34KB
[19:39:28]   dist/v0/amp-beopinion-0.1.mjs: 5.07KB
[19:39:28]   dist/v0/amp-bind-0.1.mjs: 15.09KB
[19:39:28]   dist/v0/amp-bodymovin-animation-0.1.mjs: 5.78KB
[19:39:28]   dist/v0/amp-brid-player-0.1.mjs: 8.94KB
[19:39:28]   dist/v0/amp-brightcove-0.1.mjs: 8.73KB
[19:39:28]   dist/v0/amp-byside-content-0.1.mjs: 4.83KB
[19:39:28]   dist/v0/amp-call-tracking-0.1.mjs: 2.32KB
[19:39:28]   dist/v0/amp-carousel-0.1.mjs: 8.14KB
[19:39:28]   dist/v0/amp-carousel-0.2.mjs: 8.6KB
[19:39:28]   dist/v0/amp-connatix-player-0.1.mjs: 1.12KB
[19:39:28]   dist/v0/amp-consent-0.1.mjs: 10.7KB
[19:39:28]   dist/v0/amp-crypto-polyfill-0.1.mjs: 3.75KB
[19:39:28]   dist/v0/amp-dailymotion-0.1.mjs: 8.53KB
[19:39:28]   dist/v0/amp-date-countdown-0.1.mjs: 2.72KB
[19:39:28]   dist/v0/amp-date-display-0.1.mjs: 1.51KB
[19:39:29]   dist/v0/amp-date-display-0.2.mjs: 7.09KB
[19:39:29]   dist/v0/amp-date-picker-0.1.mjs: 117.36KB
[19:39:29]   dist/v0/amp-delight-player-0.1.mjs: 8.68KB
[19:39:29]   dist/v0/amp-dynamic-css-classes-0.1.mjs: 1.75KB
[19:39:29]   dist/v0/amp-embedly-card-0.1.mjs: 5.05KB
[19:39:29]   dist/v0/amp-experiment-0.1.mjs: 1.98KB
[19:39:29]   dist/v0/amp-experiment-1.0.mjs: 6.78KB
[19:39:29]   dist/v0/amp-facebook-0.1.mjs: 5.7KB
[19:39:29]   dist/v0/amp-facebook-comments-0.1.mjs: 5.69KB
[19:39:29]   dist/v0/amp-facebook-like-0.1.mjs: 5.31KB
[19:39:29]   dist/v0/amp-facebook-page-0.1.mjs: 5.7KB
[19:39:29]   dist/v0/amp-fit-text-0.1.mjs: 1.29KB
[19:39:29]   dist/v0/amp-font-0.1.mjs: 1.76KB
[19:39:30]   dist/v0/amp-form-0.1.mjs: 13.72KB
[19:39:30]   dist/v0/amp-fx-collection-0.1.mjs: 4.86KB
[19:39:30]   dist/v0/amp-fx-flying-carpet-0.1.mjs: 1.76KB
[19:39:30]   dist/v0/amp-geo-0.1.mjs: 3.6KB
[19:39:30]   dist/v0/amp-gfycat-0.1.mjs: 8.02KB
[19:39:30]   dist/v0/amp-gist-0.1.mjs: 4.9KB
[19:39:30]   dist/v0/amp-google-document-embed-0.1.mjs: 1.03KB
[19:39:30]   dist/v0/amp-google-vrview-image-0.1.mjs: 2.06KB
[19:39:30]   dist/v0/amp-gwd-animation-0.1.mjs: 3.44KB
[19:39:30]   dist/v0/amp-hulu-0.1.mjs: 0.86KB
[19:39:30]   dist/v0/amp-iframe-0.1.mjs: 7.88KB
[19:39:30]   dist/v0/amp-ima-video-0.1.mjs: 11.25KB
[19:39:30]   dist/v0/amp-image-lightbox-0.1.mjs: 8.63KB
[19:39:30]   dist/v0/amp-image-slider-0.1.mjs: 5.03KB
[19:39:30]   dist/v0/amp-image-viewer-0.1.mjs: 7.69KB
[19:39:30]   dist/v0/amp-imgur-0.1.mjs: 1.08KB
[19:39:30]   dist/v0/amp-inline-gallery-0.1.mjs: 3.08KB
[19:39:30]   dist/v0/amp-inputmask-0.1.mjs: 21.73KB
[19:39:30]   dist/v0/amp-instagram-0.1.mjs: 1.79KB
[19:39:30]   dist/v0/amp-install-serviceworker-0.1.mjs: 3.38KB
[19:39:30]   dist/v0/amp-izlesene-0.1.mjs: 1.15KB
[19:39:30]   dist/v0/amp-jwplayer-0.1.mjs: 1.38KB
[19:39:30]   dist/v0/amp-kaltura-player-0.1.mjs: 1.4KB
[19:39:30]   dist/v0/amp-lightbox-0.1.mjs: 6.71KB
[19:39:30]   dist/v0/amp-lightbox-gallery-0.1.mjs: 17.03KB
[19:39:30]   dist/v0/amp-link-rewriter-0.1.mjs: 2.72KB
[19:39:30]   dist/v0/amp-list-0.1.mjs: 11.55KB
[19:39:30]   dist/v0/amp-live-list-0.1.mjs: 6.93KB
[19:39:30]   dist/v0/amp-loader-0.1.mjs: 3.63KB
[19:39:30]   dist/v0/amp-login-done-0.1.mjs: 7.25KB
[19:39:30]   dist/v0/amp-mathml-0.1.mjs: 5.04KB
[19:39:30]   dist/v0/amp-mega-menu-0.1.mjs: 3.64KB
[19:39:30]   dist/v0/amp-megaphone-0.1.mjs: 1.52KB
[19:39:30]   dist/v0/amp-minute-media-player-0.1.mjs: 8.39KB
[19:39:30]   dist/v0/amp-mowplayer-0.1.mjs: 8.28KB
[19:39:30]   dist/v0/amp-mraid-0.1.mjs: 2KB
[19:39:30]   dist/v0/amp-mustache-0.1.mjs: 13.7KB
[19:39:30]   dist/v0/amp-mustache-0.2.mjs: 12.24KB
[19:39:30]   dist/v0/amp-nested-menu-0.1.mjs: 2.86KB
[19:39:30]   dist/v0/amp-next-page-0.1.mjs: 13.18KB
[19:39:30]   dist/v0/amp-next-page-1.0.mjs: 13.14KB
[19:39:30]   dist/v0/amp-nexxtv-player-0.1.mjs: 8.66KB
[19:39:30]   dist/v0/amp-o2-player-0.1.mjs: 1.01KB
[19:39:30]   dist/v0/amp-ooyala-player-0.1.mjs: 8.14KB
[19:39:30]   dist/v0/amp-orientation-observer-0.1.mjs: 2.26KB
[19:39:30]   dist/v0/amp-pan-zoom-0.1.mjs: 8.14KB
[19:39:30]   dist/v0/amp-pinterest-0.1.mjs: 6.17KB
[19:39:30]   dist/v0/amp-playbuzz-0.1.mjs: 5.68KB
[19:39:30]   dist/v0/amp-position-observer-0.1.mjs: 3.41KB
[19:39:30]   dist/v0/amp-powr-player-0.1.mjs: 8.61KB
[19:39:30]   dist/v0/amp-reach-player-0.1.mjs: 0.77KB
[19:39:30]   dist/v0/amp-recaptcha-input-0.1.mjs: 7.95KB
[19:39:30]   dist/v0/amp-redbull-player-0.1.mjs: 8.06KB
[19:39:31]   dist/v0/amp-reddit-0.1.mjs: 5.04KB
[19:39:31]   dist/v0/amp-riddle-quiz-0.1.mjs: 1.02KB
[19:39:31]   dist/v0/amp-script-0.1.mjs: 18.28KB
[19:39:31]   dist/v0/amp-script-worker-0.1.js: 13.29KB
[19:39:31]   dist/v0/amp-selector-0.1.mjs: 3.46KB
[19:39:31]   dist/v0/amp-share-tracking-0.1.mjs: 1.88KB
[19:39:31]   dist/v0/amp-sidebar-0.1.mjs: 7.4KB
[19:39:31]   dist/v0/amp-sidebar-0.2.mjs: 7.46KB
[19:39:31]   dist/v0/amp-skimlinks-0.1.mjs: 6.56KB
[19:39:31]   dist/v0/amp-slides-0.1.mjs: 0.44KB
[19:39:31]   dist/v0/amp-smartlinks-0.1.mjs: 4.84KB
[19:39:31]   dist/v0/amp-social-share-0.1.mjs: 5.4KB
[19:39:31]   dist/v0/amp-social-share-0.2.mjs: 10.57KB
[19:39:31]   dist/v0/amp-soundcloud-0.1.mjs: 0.96KB
[19:39:31]   dist/v0/amp-springboard-player-0.1.mjs: 1.11KB
[19:39:31]   dist/v0/amp-standalone-0.1.mjs: 1.26KB
[19:39:31]   dist/v0/amp-sticky-ad-0.1.mjs: 2.35KB
[19:39:31]   dist/v0/amp-sticky-ad-1.0.mjs: 2.35KB
[19:39:31]   dist/v0/amp-story-0.1.mjs: 45.11KB
[19:39:32]   dist/v0/amp-story-1.0.mjs: 73.26KB
[19:39:32]   dist/v0/amp-story-auto-ads-0.1.mjs: 12.68KB
[19:39:32]   dist/v0/amp-story-education-0.1.mjs: 4.06KB
[19:39:32]   dist/v0/amp-stream-gallery-0.1.mjs: 8.59KB
[19:39:32]   dist/v0/amp-subscriptions-0.1.mjs: 18.7KB
[19:39:32]   dist/v0/amp-subscriptions-google-0.1.mjs: 35.86KB
[19:39:32]   dist/v0/amp-timeago-0.1.mjs: 5.81KB
[19:39:32]   dist/v0/amp-timeago-0.2.mjs: 11.51KB
[19:39:32]   dist/v0/amp-truncate-text-0.1.mjs: 3.18KB
[19:39:32]   dist/v0/amp-twitter-0.1.mjs: 5.67KB
[19:39:32]   dist/v0/amp-user-notification-0.1.mjs: 4.07KB
[19:39:32]   dist/v0/amp-video-0.1.mjs: 9.15KB
[19:39:32]   dist/v0/amp-video-docking-0.1.mjs: 9.96KB
[19:39:32]   dist/v0/amp-video-iframe-0.1.mjs: 8.93KB
[19:39:32]   dist/v0/amp-viewer-assistance-0.1.mjs: 2.22KB
[19:39:32]   dist/v0/amp-viewer-integration-0.1.mjs: 6.21KB
[19:39:32]   dist/v0/amp-vimeo-0.1.mjs: 8.09KB
[19:39:32]   dist/v0/amp-vine-0.1.mjs: 0.8KB
[19:39:32]   dist/v0/amp-viqeo-player-0.1.mjs: 10.49KB
[19:39:33]   dist/v0/amp-viz-vega-0.1.mjs: 139.19KB
[19:39:33]   dist/v0/amp-vk-0.1.mjs: 1.75KB
[19:39:33]   dist/v0/amp-web-push-0.1.mjs: 5.62KB
[19:39:33]   dist/v0/amp-wistia-player-0.1.mjs: 8.17KB
[19:39:33]   dist/v0/amp-yotpo-0.1.mjs: 5.04KB
[19:39:33]   dist/v0/amp-youtube-0.1.mjs: 9.09KB
[19:39:33] Finished 'bundle-size' after 8.68 s
rsimha@rsimha-macbookpro2 (2020-04-09-BabelRefactorRedux):~/src/amphtml$ 
```

Co-authored by @kristoferbaxter. Many thanks to him for the debugging help.

### CODE REVIEW

**Comment by rsimha** [comment]
@kristoferbaxter @erwinmombay @jridgewell To review only the changes added by this PR, see [these commits](https://github.com/ampproject/amphtml/pull/27683/files/775d22de0228c1bc8699d1c20e655f3057eb14a3..39061a270f219ba865dde98b1c6914baece8d84a).

File: build-system/babel-config/single-pass-config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+/**
+ * Gets the config for babel transforms run while building the dependency tree
+ * during `gulp dist --single_pass`.
+ *
+ * @return {!Object}
+ */
+function getSinglePassDepsConfig() {
+  const singlePassDepsConfig = getPreClosureConfig();
```
Comment (kristoferbaxter) [inline build-system/babel-config/single-pass-config.js:45]: Let's open a PR to address this pattern in the near future, or at the minimum add a comment about this intentionally being a clone of the value.

Comment (rsimha) [reply build-system/babel-config/single-pass-config.js:46]: Added a comment.

**Comment by rsimha** [comment]
@dvoytenko FYI, the way `amp-viewer-host` was being closure-compiled all this time (with `WHITESPACE_ONLY`), broke `gulp dist --esm`. I've moved the file to a regular closure compilation for now to unblock us.

If the file shouldn't be closure compiled, we should remove it from [`build-system/compile/bundles.config.js`](https://github.com/ampproject/amphtml/blob/7ec9a750de6d6058586a20c2d955f425b2306158/build-system/compile/bundles.config.js#L123-L135), and instead directly copy it to `dist/`.

Let me know your preference. Happy to follow up with a separate PR that does this.

**Comment by rsimha** [comment]
For posterity, here are the output diffs before / after this PR:

image

- Outputs for `gulp build`, `gulp dist`, `gulp dist --fortesting`, and `gulp --compiled` are byte-for-byte identical
- `v0.js` built by `gulp dist --single_pass` is off by 1 byte because of slightly different mangles names
- `v0.mjs` built by `gulp dist --esm` is off by slightly more because we were not applying esm-only plugins to all files during esm builds (this is fixed now)

**Pull request merged**